CONTRIBUTION.md

```diff
@@ -28,58 +28,126 @@ We know that as of 1.12 this is not the case for great parts of the mod, but we
 * Avoid non-private variables in API-Code and provide getters/setters instead. Bear in mind that this does not apply to constants (`public static final Type xyz = ...`) though.
 * In general try to use the smallest visibility-scope possible, expcept if it's likely that others will want access to the given method.
-* enforce Immutability where possible
-    * use ImmutableTypes for immutable values (`ImmutableList`, `ImmutableSet`, `ImmutableMap`, etc.)
-    * avoid public properties
-    
+* Enforce immutability where possible
+  * Use immutablet types for immutable values (`ImmutableList`, `ImmutableSet`, `ImmutableMap`, etc.)
+  * Avoid public properties
+* Always use capital letters for format suffixes and literals
+  * Bad format suffix: `0.5f`, `1000l`
+  * Good format suffiz: `0.5F`, `1000L`
+  * Bad iteral: `0xfff`, `0xbeef`
+  * Good literal: `0xFFF`, `0xBEEF`.
+* No nested classes with 2 levels or deeper.
+
 ### Packages
 * Don't use packages like there is a limited allowed amount, for the sake of code readability and maintainabilty make as may packages as logically makes sense. Attempt to group as much common functionality as possible within a package. Use our `common.utils` package as an example of functionality grouping. 
-* Respect the 3 base packages: `common`, `client` and `api`. If you ever feel that something doesn't fit in these packages then you're likely thinking about it wrong. 
-  - `common` is used for everything that isn't `client` specific or part of our `api`.
-  - `client` is used for everything that isn't common funcionality and is purely for the client side of the mod.
-  - `api` our api package is for all api related functionality, it's rare you should be using this package unless you're specifically working on api specific additions.
+* Respect the 3 base packages: `common`, `client` and `api`. If you ever feel that something doesn't fit in these packages then you're likely thinking about it wrong:
+  * `common` is used for everything that isn't `client` specific or part of our `api`.
+  * `client` is used for everything that isn't common funcionality and is purely for the client side of the mod.
+  * `api` our api package is for all api related functionality, it's rare you should be using this package unless you're specifically working on api specific additions.
 * Use sensible and logical naming for your packages and avoid prefixing or sufixing your package names. For example `common.items.gadgets` is good, `common.itemGadgets` is bad as a sub-package should have been used. 
-* Always use lower case and respect the official Java guide to creating packages which can be found [here](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html)
+* Always use lower case and respect the official Java guide to creating packages which can be found [here](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html).
+
+### Class Structure
```

Comment:
No, this is just standards and doesn't need to be re-enforced. We want people to read this doc, not skim it.  also static methods before the constuctor is awful.

Reply:
I actually always do static Methods before the constructor because I think it's aweful to mix static and non static Methods... XD
Though I agree that people should not want to skip the doc... :thumpsup:

Reply:
Monster lol